rust/zypp-agama/src/lib.rs

```diff
@@ -64,36 +65,90 @@ where
     Some(progress_callback::<F>)
 }
 
-pub fn init_target<F>(root: &str, progress: F) -> ZyppResult<()>
+// Mutex for the Zypp C API which is not thread safe.
+// TODO: guard some (global) context instead of ().
+//static ZYPP_MUTEX: Mutex<*mut zypp_agama_sys::Zypp> = Mutex::new(std::ptr::null_mut());
+static ZYPP_MUTEX: Mutex<()> = Mutex::new(());
+pub struct Zypp<'a> {
+    // underscore prevents
+    //     warning: field `guard` is never read
+    _guard: MutexGuard<'a, ()>,
+    // stupid spelling, attempt to prevent private access
+    in_ner: *mut zypp_agama_sys::Zypp,
```

Comment:
I hate to admit, `pub` visibility rules still elude me. I thought having a (not `pub`) `inner` field would prevent _anyone_ using `zypp.inner` but apparently I need a module boundary for that?

Reply:
if you leave it private, then it means that only imp of Struct can access it...or where it needs that module boundary?

Reply:
in general it is well written here https://doc.rust-lang.org/rust-by-example/mod/struct_visibility.html .. so it is hidden when used outside of module.

Reply:
and copy from IRC: ideally this inner should not be pointer used everywhere, but something rust safe, then is in `fn inner` casted to unsafe pointer.

Reply:
and maybe even distinguish where we need to pass mut pointer and where const is enough.